From b86e4d24420def87fd94922ba0a2be945165323d Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Tue, 6 Feb 2018 11:26:15 +0200
Subject: [PATCH 3347/5242] MLK-17517: ARM64: dts: imx8mq-evk: Fix record for
 more than two channels

commit  8feab278f98d656c45f59c2251815ddc984c4f29 from
https://source.codeaurora.org/external/imx/linux-imx.git

SAI5_RXDn must be muxed to RX_DATAn. For n = {1, 2, 3}
this mapping was not correctly done so only channels
recorded from data line 0 were seen in .wav capture.

Suggested-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Reviwed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 2547352..316cf5c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -370,9 +370,9 @@
 				MX8MQ_IOMUXC_SAI5_RXC_SAI5_RX_BCLK	0xd6
 				MX8MQ_IOMUXC_SAI5_RXFS_SAI5_RX_SYNC	0xd6
 				MX8MQ_IOMUXC_SAI5_RXD0_SAI5_RX_DATA0	0xd6
-				MX8MQ_IOMUXC_SAI5_RXD1_SAI5_TX_SYNC	0xd6
-				MX8MQ_IOMUXC_SAI5_RXD2_SAI5_TX_BCLK	0xd6
-				MX8MQ_IOMUXC_SAI5_RXD3_SAI5_TX_DATA0	0xd6
+				MX8MQ_IOMUXC_SAI5_RXD1_SAI5_RX_DATA1    0xd6
+				MX8MQ_IOMUXC_SAI5_RXD2_SAI5_RX_DATA2    0xd6
+				MX8MQ_IOMUXC_SAI5_RXD3_SAI5_RX_DATA3    0xd6
 			>;
 		};
 
-- 
1.7.9.5

